### PR TITLE
12 footer block

### LINKED ISSUE #12: Footer
- Three lists of links
- Replace icons for Twitter ([check X brand toolkit](https://about.twitter.com/en/who-we-are/brand-toolkit)) and YouTube
- **To do:** check what happens to current links "Cookies" and "Do Not Sell or Share My Personal Information"

### PR DESCRIPTION
**Pull request by avinashNChavan**
Fix #12 

Test URLs:
- Before: https://main--vg-macktrucks-com-rd--netcentric.hlx.page/
- After: https://12-footer-block--vg-macktrucks-com-rd--netcentric.hlx.page/

### CODE REVIEW

**Comment by cogniSyb** [comment]
The URL provided doesn’t show the new footer in place. When I checkout the branch locally, it’s the same. What am I missing here?

I’ll do a bit of code review, but I can finish it once I got the change viewable in a browser.

File: blocks/v2-footer/v2-footer.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  footer.innerHTML = html;
+
+  // // adding the 'block' class and 'data-block-name'
+  // // needed to load blocks
+  const eloquaFormEl = footer.querySelector('.eloqua-form');
+  eloquaFormEl?.classList.add('block');
+  eloquaFormEl?.setAttribute('data-block-name', 'eloqua-form');
+
+  // transforming icons into font-awesome
+  footer.querySelectorAll('span.icon').forEach((icon) => {
```
Comment (cogniSyb) [inline blocks/v2-footer/v2-footer.js:32]: We’ll replace FontAwesome gradually in the project for SVG icons. Could you make sure we’ll have an SVG icon for all social media? Make sure to implement the [latest one for Twitter](https://about.twitter.com/en/who-we-are/brand-toolkit).

File: blocks/v2-footer/v2-footer.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        <div class="v2-footer-form-section v2-form">
+            <div class="form v2-footer-form-header">
+              ${fourthHeader?.outerHTML}
+            </div>
+            <div class="form footer-form">
+              ${formSection?.outerHTML}
+            </div>
+          </div>
+      </div>
+      <hr />
```
Comment (cogniSyb) [inline blocks/v2-footer/v2-footer.js:97]: Replace the `<hr /> with styling for `.footer-bottom-section`

File: blocks/v2-footer/v2-footer.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        const submitButton = block.querySelector('input[type="submit"]');
+        const emailInput = block.querySelector('input[name="emailAddress"]');
+        const label = emailInput.parentElement.querySelector('label');
+        const emailAndSubmitContainer = createElement('span', ['email-and-submit-container']);
+
+        // change the submit button to arrow button
+        // and display it sticked to the right side of email input
+        if (submitButton && emailInput) {
+          const parent = emailInput.parentElement;
+          submitButton.value = '→';
```
Comment (cogniSyb) [inline blocks/v2-footer/v2-footer.js:142]: Should be replaced with actual text which is visually hidden. The button should contain the SVG icon.

File: blocks/v2-footer/v2-footer.css [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+    border-bottom: 2px solid;
+    line-height: 22px;
+    margin-top: 0;
+  }
+  
+  .v2-footer ul li a {
+    color: var(--c-secondary-silver);
+  }
+  
+  .v2-footer .v2-footer-bottom-section li a {
```
Comment (cogniSyb) [inline blocks/v2-footer/v2-footer.css:60]: Combine this selector with the one above it

**Review by cogniSyb — CHANGES_REQUESTED** [review]

File: blocks/v2-footer/v2-footer.js (hunk 2 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js]: Updated

File: blocks/v2-footer/v2-footer.js (hunk 3 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js]: Updated

File: blocks/v2-footer/v2-footer.css (hunk 4 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css:66]: Updated

**Review by kesiah — CHANGES_REQUESTED** [review]
Could we change the name of the svg `arrow.svg` to `form-submit-arrow.svg`?

**Comment by cogniSyb** [comment]
Could you add links to all items in the footer? "Sustainability" and "Brand Partnerships" don’t have an `<a>`

File: blocks/v2-footer/v2-footer.css [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+  
+
+  .v2-social { 
+    grid-area: v2-social;
+    margin-top: 0px;
+  }
+
+  .v2-content { grid-area: v2-content; }
+
+  .v2-form {
```
Comment (cogniSyb) [inline blocks/v2-footer/v2-footer.css:159]: The eloqua-form.css file sets `max-width: 950px;`. From around 1024px viewport until the breakpoint, this causes the form to not be aligned to the left. Could you fix that?

File: blocks/v2-footer/v2-footer.css [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+  
+  .v2-footer .elq-form .item-padding {
+    padding: 0;
+  }
+  
+  .v2-footer .eloqua-form .elq-form input[type="text"] {
+    background-color: var(--c-primary-white);
+    padding: 15px 10px;
+    border: none;
+    width: 200px;
```
Comment (cogniSyb) [inline blocks/v2-footer/v2-footer.css:269]: In the design for mobile, the form field is wider. There it’s 295px, but it should be 100% width minus the button. I think a `max-width: 320px` would be a good size.

File: icons/arrow.svg [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<svg width="48" height="48" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
+<g id="Frame 2609178">
+<rect width="48" height="48" fill="#B3976B"/>
+<path id="Vector 65" d="M14 23C13.4477 23 13 23.4477 13 24C13 24.5523 13.4477 25 14 25V23ZM32.7071 24.7071C33.0976 24.3166 33.0976 23.6834 32.7071 23.2929L26.3431 16.9289C25.9526 16.5384 25.3195 16.5384 24.9289 16.9289C24.5384 17.3195 24.5384 17.9526 24.9289 18.3431L30.5858 24L24.9289 29.6569C24.5384 30.0474 24.5384 30.6805 24.9289 31.0711C25.3195 31.4616 25.9526 31.4616 26.3431 31.0711L32.7071 24.7071ZM14 25L32 25V23L14 23V25Z" fill="#1D1D1D"/>
```
Comment (cogniSyb) [inline icons/arrow.svg:4]: Can you export the icon without the background? We need the button to be styled with different states, as you can find here: https://www.figma.com/file/URJhT8pkntqc6h6v5wSLqx/Mack-Core-Design-Kit?type=design&node-id=49-4333&mode=dev

File: icons/yt.svg [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
```
Comment (cogniSyb) [inline icons/yt.svg:1]: These SVG files should be cleaned up with SVGO. I’ll resize them all for you and will share them :)

**Review by cogniSyb — CHANGES_REQUESTED** [review]

File: blocks/v2-footer/v2-footer.css [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+  
+  .v2-footer .v2-footer-social-media:hover {
+    color: var(--c-tertiary-cool-white);
+  }
+  
+  .v2-footer .v2-footer-social-media::before {
+    width: 17px;
+  }
+  
+  .v2-footer h1,
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.css [hunk 10]
```diff
@@ -0,0 +1,393 @@
...
+  .v2-footer .v2-footer-bottom-section li a {
+    color: var(--c-secondary-silver);
+  }
+  
+  .v2-footer ul li a:hover {
+    color: var(--c-secondary-white);
+    text-decoration: none;
+  }
+  
+  .v2-footer h1 ~ ul,
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.css [hunk 11]
```diff
@@ -0,0 +1,393 @@
...
+    display: flex;
+    justify-content: space-between;
+    flex-wrap: wrap;
+  }
+
+  .v2-footer .v2-footer-links-section ul {
+    min-width: 50%;
+  }
+  
+  .v2-footer .v2-footer-form-header h3{
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.css [hunk 12]
```diff
@@ -0,0 +1,393 @@
...
+    gap: 0 0; 
+    /* stylelint-disable-next-line declaration-block-no-redundant-longhand-properties */
+    grid-template-areas: 
+      "v2-logo"
+      "v2-content"
+      "v2-form"
+      "v2-social"; 
+      height: 70vh;
+  }
+
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css:142]: Fixed

File: blocks/v2-footer/v2-footer.css [hunk 13]
```diff
@@ -0,0 +1,393 @@
...
+    grid-area: v2-form;
+    height: 15vh;
+  }
+
+  .v2-logo { 
+    grid-area: v2-logo; 
+    height: 15vh;
+  }
+
+  @media (min-width: 495px) {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.css (hunk 5 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css:150]: Updated

File: blocks/v2-footer/v2-footer.css (hunk 6 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: icons/arrow.svg (hunk 7 above)
Comment (avinashNChavan) [reply icons/arrow.svg]: Updated

File: icons/yt.svg (hunk 8 above)
Comment (avinashNChavan) [reply icons/yt.svg]: Fixed

**Comment by cogniSyb** [comment]
As [shared by the Volvo PO](https://github.com/Netcentric/vg-volvotrucks-us-rd/issues/50), we should add the legal link back in. Could you add [Do Not Sell or Share My Personal Information](https://www.volvogroup.com/en/tools/privacy/data-subject-rights.html)?

File: blocks/v2-footer/v2-footer.css [hunk 14]
```diff
@@ -0,0 +1,413 @@
...
+      gap: 20px;
+      flex-wrap: nowrap;
+    }
+
+    .v2-footer .v2-footer-links-section ul {
+      min-width: 0;
+      width:160px;
+    }
+
+    .v2-footer .v2-footer-links-section ul li{
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.css:188]: It has the same css rule at line 160, this is redundant .
Also gap between each link is `16px` 
Refer here https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?node-id=2243%3A159264&mode=dev

File: blocks/v2-footer/v2-footer.css [hunk 15]
```diff
@@ -0,0 +1,413 @@
...
+  }
+  
+  .confirm-checkbox.show {
+    opacity: 1;
+    visibility: visible;
+  }
+  
+  /* eloqua footer form styles - END */
+  
+
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.css:394]: Extra line

File: blocks/v2-footer/v2-footer.js [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+  const bottomLinksList = [...footer.querySelectorAll('ul')].at(-1);
+  bottomLinksList?.classList.add('v2-footer-bottom-section-content');
+
+  const formSection = footer.querySelector('.eloqua-form');
+  formSection?.classList.add('v2-footer-form-section');
+
+  const footerTemplate = `
+    <div class="v2-footer-content">
+      <div class="v2-footer-main-content">
+        <div class="v2-footer-logo-section v2-logo">
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.js:81]: Why do we need 2 class names here?

File: blocks/v2-footer/v2-footer.js [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+          ${logoLink?.outerHTML}
+        </div>
+        <div class="v2-footer-social-section v2-social">
+          ${socialMediaSection?.outerHTML}
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.js:85]: Same as above ?

File: blocks/v2-footer/v2-footer.js [hunk 18]
```diff
@@ -0,0 +1,210 @@
...
+        </div>
+        <div class="v2-footer-links-section v2-content">
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.js:87]: Same here

File: blocks/v2-footer/v2-footer.js [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+          ${firstLinks?.outerHTML}
+          ${secondLinks?.outerHTML}
+          ${thirdLinks?.outerHTML}
+        </div>
+        <div class="v2-footer-form-section v2-form">
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.js:92]: Same here

File: blocks/v2-footer/v2-footer.css [hunk 20]
```diff
@@ -0,0 +1,413 @@
...
+      gap: 0 0; 
+      /* stylelint-disable-next-line declaration-block-no-redundant-longhand-properties */
+      grid-template-areas: 
+      "v2-logo v2-content v2-form"
+      "v2-social v2-content v2-form";
+      height: 15vh;
+    }
+
+    .v2-social { 
+      margin-top: -80px;
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.css:235]: Unnecessary negative margin. You can get rid of `height: 15vh` for `v2-logo` to solve the problem of positioning elements

File: blocks/v2-footer/v2-footer.css [hunk 21]
```diff
@@ -0,0 +1,413 @@
...
+    line-height: 22px;
+  }
+  
+  .v2-footer ul li a,
+  .v2-footer .v2-footer-bottom-section li a {
+    color: var(--c-secondary-silver);
+  }
+  
+  .v2-footer ul li a:hover {
+    color: var(--c-secondary-white);
```
Comment (Lakshmishri) [inline blocks/v2-footer/v2-footer.css:67]: This variable isn't defined, so i can't see a hover state

**Review by Lakshmishri — CHANGES_REQUESTED** [review]
- I think there are way too many links
    Screenshot 2023-08-17 at 16 18 37

 
- Mobile view looks completely off
   Screenshot 2023-08-17 at 16 47 28

- Can you rename the `form-submit-arrow.svg` to `arrow-right.svg`
- Can you put back the footer function to franklin lib, it still stays there and we wouldn't use that

**Review by kesiah — CHANGES_REQUESTED** [review]
Make sure to change the fill attribute from SVG's files to `currentColor` ;)

File: blocks/v2-footer/v2-footer.css (hunk 14 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.css (hunk 15 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.js (hunk 16 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js]: Fixed

File: blocks/v2-footer/v2-footer.js (hunk 17 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js:99]: Fixed

File: blocks/v2-footer/v2-footer.js (hunk 18 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.js]: Fixed

File: blocks/v2-footer/v2-footer.css (hunk 20 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css:327]: Updated

File: blocks/v2-footer/v2-footer.css (hunk 21 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

**Comment by avinashNChavan** [comment]
> Make sure to change the fill attribute from SVG's files to `currentColor` ;)

Updated

**Review by Lakshmishri — CHANGES_REQUESTED** [review]
There is a lot of gap between the logo and links in mobile. Check the design [here](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?node-id=2243%3A160030&mode=dev)

Lot of spacing between the rows compared to design
Screenshot 2023-08-18 at 15 05 54

Screenshot 2023-08-18 at 15 12 38
there is no space between them 

Screenshot 2023-08-18 at 15 15 02

You can see that they are not aligned

Accordion icon missing in smaller screen size

File: blocks/v2-footer/v2-footer.css [hunk 22]
```diff
@@ -0,0 +1,590 @@
...
+  
+  .v2-footer .v2-footer-social-media-section {
+    margin-top: 21px;
+    display: flex;
+    gap: 20px;
+    flex-wrap: wrap;
+  }
+  
+  .v2-footer .v2-footer-social-media {
+    border-radius: 17.5px;
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:31]: Not sure why `border-radius`, `display`, `font-size`, `text-decoration` is necessary. 🤔 
I would say that colour, width and height is only necessary.

File: blocks/v2-footer/v2-footer.css [hunk 23]
```diff
@@ -0,0 +1,590 @@
...
+      border-bottom: none;
+      justify-content: center;
+      gap: 20px;
+    }
+
+    .v2-footer .v2-pre-footer-content ul li::after {
+      content: '';
+      width: 1px;
+      height: 64px;
+      background: black;
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:381]: please use variables :)

File: blocks/v2-footer/v2-footer.css [hunk 24]
```diff
@@ -0,0 +1,590 @@
...
+      top: 2px;
+      right: -30px;
+    }
+
+    .v2-footer .v2-pre-footer-list-wrapper {
+      padding: 24px 135px;
+    }
+
+    .v2-footer .v2-pre-footer-content ul li {
+      width: 33.33%;
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:368]: you can also use `calc(100% / 3)`

File: blocks/v2-footer/v2-footer.css [hunk 25]
```diff
@@ -0,0 +1,590 @@
...
+
+    /* stylelint-disable-next-line no-duplicate-selectors */
+    .v2-footer .v2-footer-main-content{
+      display: grid; 
+      grid-template-columns: 1fr 1fr 1fr; 
+      grid-template-rows: 1fr 1fr; 
+      gap: 0 0; 
+      /* stylelint-disable-next-line declaration-block-no-redundant-longhand-properties */
+      grid-template-areas: 
+      "v2-logo v2-content v2-form"
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:334]: Use single quotes and be careful with the indentation in all the document.

File: blocks/v2-footer/v2-footer.css [hunk 26]
```diff
@@ -0,0 +1,590 @@
...
+    display: flex;
+    width: 100%;
+    padding: 24px;
+    justify-content: flex-start;
+    align-items: baseline;
+    background-color: var(--c-secondary-graphite);
+  }
+
+  .v2-footer .v2-pre-footer-list-wrapper h3 {
+    color: var(--primary-white-ffffff, #FFF);
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:222]: where is this colour coming from? `--primary-white-ffffff` I think that we should use `--c-primary-white`, check in styles please

File: blocks/v2-footer/v2-footer.css [hunk 27]
```diff
@@ -0,0 +1,590 @@
...
+  .v2-footer .v2-pre-footer-content ul li {
+    font-size: 24px;
+    font-style: normal;
+    font-weight: 500;
+    line-height: 140%; /* 33.6px */
+    letter-spacing: 0.48px;
+    color:var(--c-primary-black);
+    position: relative;
+    width: 100%;
+    border-bottom: solid black 1px;
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:193]: Use the color variables

File: blocks/v2-footer/v2-footer.css [hunk 28]
```diff
@@ -0,0 +1,590 @@
...
+  }
+
+  .v2-footer .v2-footer-main-content {
+    display: grid; 
+    grid-template-columns: 1fr; 
+    grid-template-rows: 1fr 1fr 1fr 1fr; 
+    gap: 0 0; 
+    /* stylelint-disable-next-line declaration-block-no-redundant-longhand-properties */
+    grid-template-areas: 
+      "v2-logo"
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:144]: Single quotes, thanks!

File: blocks/v2-footer/v2-footer.css [hunk 29]
```diff
@@ -0,0 +1,590 @@
...
+  }
+  
+  .v2-footer .v2-footer-form-header .v2-footer__title {
+    font-family: var(--ff-body);
+    font-size: 12px;
+    font-style: normal;
+    font-weight: 700;
+    line-height: 16px;
+    letter-spacing: 1.92px;
+    text-transform: uppercase;
```
Comment (kesiah) [inline blocks/v2-footer/v2-footer.css:133]: @Lakshmishri correct me if I'm wrong, but shouldn't the editors decide if the text is in uppercase?

**Review by kesiah — CHANGES_REQUESTED** [review]

File: blocks/v2-footer/v2-footer.css [hunk 30]
```diff
@@ -0,0 +1,569 @@
...
+  }
+
+  .v2-footer .v2-pre-footer .v2-pre-footer-content ul li a {
+    color: var(--c-primary-black);
+    position: relative;
+    display: inline-block;
+    width: 100%;
+  }
+
+  .v2-footer .v2-pre-footer .v2-pre-footer-content ul li a::before {
```
[the remark this answers is not in the document]
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.css (hunk 22 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.css (hunk 23 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.css (hunk 24 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.css (hunk 25 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.css (hunk 26 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.css (hunk 27 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Fixed

File: blocks/v2-footer/v2-footer.css (hunk 28 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

File: blocks/v2-footer/v2-footer.css (hunk 29 above)
Comment (avinashNChavan) [reply blocks/v2-footer/v2-footer.css]: Updated

**Pull request closed without merging**